### ISSUE TITLE
Unable to start GUI application with "CreateWndow(): rdp_peer is not initalized"

### ISSUE THREAD

**Issue by shibukawa** [microsoft/wslg#125]
I tried on two computers. One computer works perfectly (Dell XPS12), but another one doesn't (ASUS Zephyrus G14).

# Environment

```none
Windows build number: [run `[Environment]::OSVersion` for powershell, or `ver` for cmd]
Your Distribution version: [On Debian or Ubuntu run `lsb_release -r` in WSL]
Your WSLg version: [Open 'Apps and Features' and check the version of 'Windows Subsystem for Linux GUI app support', e.g: 0.2.3.13]
```

Host OS: Microsoft Windows [Version 10.0.21364.1000]
Guest OS: Ubuntu 20.04, 21.04
PC: ASUS Zephyrus G14 (Ryzen + GeForce RTX 2060)

```
shibukawa@LAPTOP-MM4JT0DC:/mnt/c/Users/yoshi$ echo $DISPLAY
:0
shibukawa@LAPTOP-MM4JT0DC:/mnt/c/Users/yoshi$ l -la /tmp/.X11-unix
lrwxrwxrwx 1 root root 19 Apr 29 09:00 /tmp/.X11-unix -> /mnt/wslg/.X11-unix/
shibukawa@LAPTOP-MM4JT0DC:/mnt/c/Users/yoshi$ ls -la /tmp/.X11-unix/
total 0
drwxrwxrwx 2 root      root   60 Apr 29 09:00 .
drwxrwxrwt 5 root      root  180 Apr 29 09:00 ..
srwxrwxrwx 1 shibukawa users   0 Apr 29 09:00 X0
```

# Steps to reproduce

1. Clean WSL environments: wsl --unregister Ubuntu
2. Install Ubuntu: wsl --install -d Ubuntu
3. Create user/password at first booting
4. Run: sudo apt update && sudo apt dist-upgrade
5. Run: sudo apt install x11-apps
6. Run: xeyes or xedit

**WSL logs**: 

[weston.log](https://github.com/microsoft/wslg/files/6395554/weston.log)
[versions.txt](https://github.com/microsoft/wslg/files/6395558/versions.txt)
[pulseaudio.log](https://github.com/microsoft/wslg/files/6395613/pulseaudio.log)

#  Expected behavior

Application icon appears on task bar and application window appears on desktop.

# Actual behavior

Do nothing (no icon on taskbar, xeyes command stuck on bash)

![image](https://user-images.githubusercontent.com/564612/116487894-0aa03b80-a8cc-11eb-8417-989487c4a3e2.png)

**Comment by shibukawa** [comment]
I searched other issued and [this](https://github.com/microsoft/wslg/issues/90) has same error message. But I used prebuild kernel installed by ``wsl --update``.

**Comment by spronovo** [comment]
Thanks for your report, this one is a bit odd. It looks like mstsc.exe might have crash on the Windows host and we couldn't successfully connect Weston's RDP backend to the RDP client on Windows.

Could you shutdown wsl (wsl --shutdown from an elevated command prompt). Open task manager, go to the details tab. Launch your WSL distro again, do you see mstsc.exe appears there? does it appear and go away, come back, etc... with different PID every time?

**Comment by shibukawa** [comment]
Yes. Your are right. mstsc.exe changes its PID every three seconds.

https://user-images.githubusercontent.com/564612/116554652-699a9a80-a936-11eb-903d-bc86bc70e646.mp4

**Comment by hideyukn88** [comment]
Would you please go to Settings -> Diagnostics & feedback -> View diagnostics data -> click Open Diagnostics Data Viewer (you will need to slide to "on" first then install viewer if you haven't, it redirects to Microsoft Store app automatically). and launch Diagnostics Viewer, click "Problem reports" icon on left side bar, and find if any mstsc.exe (or remote desktop ...) shows up in middle pane. If there is select it, then the right pane will show some information, please find "Bucket ID" in there, and please share that ID with us, (and after this, you can back to "off" at Diagnostics viewer option at Settings app) thanks!

**Comment by shibukawa** [comment]
![image](https://user-images.githubusercontent.com/564612/116633174-caa48b80-a993-11eb-8804-265b02358ff0.png)

There is no report related to mstsc.exe or Remote Desktop in Problem reports pane.

![image](https://user-images.githubusercontent.com/564612/116633253-ffb0de00-a993-11eb-827e-812bb8028864.png)

Diagnostics Data pane has many data related to mstsc.exe. Do they help you?

**Comment by spronovo** [comment]
Unfortunately that's not quite the information we were looking for. Could you try this instead. Open the Feedback Hub (type "feedback in the start menu"), open a new issue and write something like below. The bug will find itself to us and we should be able to look at the crash history for your system and retrieved any crash dump from mstsc that should have gotten uploaded.

![image](https://user-images.githubusercontent.com/7545376/116650051-46232000-a935-11eb-8d46-3357206cb5d3.png)

**Comment by shibukawa** [comment]
I sent just now.

**Comment by hideyukn88** [comment]
Hi, we have received the feedback report from you and now analyzing it, once we know more or need more information from you, we will update here, thanks!

**Comment by kyaru92** [comment]
"mstsc.exe" repeated restart, keep my CPU fan spinning, and generate tons of error reports. 
I have tried "sfc /scannow" and "DISM /Online /Cleanup-Image /RestoreHealth" but no help.

**Comment by hideyukn88** [comment]
@shibukawa , I looked at the feedback from you, and there are errors logged from Hyper-V socket layer and this is preventing mstsc from connecting to WSLg, thus mstsc terminated immediately and WSLg relaunching it, this repeats.

@kyaru92 , would you please file report using Feedback Hub, too ? more reports might help us to get more clues on this, thanks!

**Comment by kyaru92** [comment]
> @shibukawa , I looked at the feedback from you, and there are errors logged from Hyper-V socket layer and this is preventing mstsc from connecting to WSLg, thus mstsc terminated immediately and WSLg relaunching it, this repeats.
> 
> @kyaru92 , would you please file report using Feedback Hub, too ? more reports might help us to get more clues on this, thanks!

at first I thought it was a system problem due to update to 21370 so I reported this problem using feedback hub yesterday. 
after read this issue i recognized that it may cause by wslg and I uninstalled that. So now only some of the previous logs are left, and I don't know which ones will help you.

**Comment by hideyukn88** [comment]
@shibukawa, @kyaru92, sorry for late to follow up. Do you still experience this issue? We have extended the logging for this issue in Feedback Hub, so if you are still seeing mstsc.exe is repeatedly restarting, would you please report using Feedback Hub. Please make sure select "Developer Platform" and "Subsystem for Linux", so proper logging will be enabled, thanks!

**Comment by shibukawa** [comment]
I tried on Windows 10 Insider Preview 21370.1 and WSLg 1.0.22, GUI applications works perfectly. Thank you.

@kyaru92 could you close this issue if you confirm on you computer?

![image](https://user-images.githubusercontent.com/564612/122941417-56450200-d3b0-11eb-8682-b33bbf8157d2.png)

**Comment by dpiet** [comment]
@spronovo @hideyukn88 I'm seeing this same behavior with latest WSL (and previous version as well):
```
WSL version: 0.61.8.0
Kernel version: 5.10.102.1
WSLg version: 1.0.39
MSRDC version: 1.2.3213
Direct3D version: 1.601.0
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.169
```
```
$ cat /mnt/wslg/versions.txt
WSLg ( x86_64 ): 1.0.39+Branch.main.Sha.cdbed82a21d382b5c74367c48a32ef4736cc8735
Mariner: VERSION="2.0.20220426"
mesa:
mesa:
pulseaudio: 2f0f0b8c3872780f15e275fc12899f4564f01bd5
FreeRDP: c574044a10003e50453acb4cf42801c5833fb572
weston: 4e99bb16dd44fc6daf4a45c46ba3e053c0637afa
```
```
$ tail /mnt/wslg/weston.log
[10:22:23.154] CreateWndow(): rdp_peer is not initalized
[10:22:23.154] CreateWndow(): rdp_peer is not initalized
[10:22:23.154] CreateWndow(): rdp_peer is not initalized
[10:22:23.154] CreateWndow(): rdp_peer is not initalized
[10:22:23.154] CreateWndow(): rdp_peer is not initalized
[10:22:23.154] CreateWndow(): rdp_peer is not initalized
[10:22:23.154] CreateWndow(): rdp_peer is not initalized
[10:22:23.182] app_list_monitor_thread: loadIconEvent is signalled. gedit
[10:22:23.182] app_list_monitor_thread: entry 0x7f9ca40023d0, image (nil)
[10:22:23.182] set_window_icon(): rdp_peer is not initalized
```

I submitted diagnostics via Feedback Hub following your earlier guidance in this issue.

**Comment by hideyukn88** [comment]
@dpiet, thanks for feedback report and I looked at it. Do you install any TerminalWorks products, such as TSPrint? If so, would you please uninstall any products from them completely and try again? It looks some DLL from that product is being loaded into msrdc.exe process (which is used with WSLg) and causing buffer overrun, and as it's detected by OS, msrdc.exe is forcibly terminated. (For internal reference, bucket: 73895a7a-72ed-176e-0176-ffaee0eb1194). If the product is essential to you, you might want to make sure it's up to date. Thanks!

**Comment by dpiet** [comment]
Yes I did have TSPrint and TSScan installed. Uninstalling TSPrint didn't help but uninstalling TSScan seems to have fixed it. Thanks! Disregard the Feedback submission, which happened after the TSPrint uninstall but before the TSScan.

**Comment by parawanderer** [comment]
Uh no idea if this is obviously the same issue or if this issue is generic, but I got this issue with the same error message OP mentioned probably because I uninstalled `xrdp` by accident. Probably via the ubuntu prompt to uninstall unneeded dependencies.

Fix was this, lol:
```bash
# inside the wsl2
sudo apt install xrdp
# make sure it's running
sudo service xrdp status
```

Windows:
```bash
wsl --shutdown
```

That's it. Pretty sure that once you manually install `xrdp` like this it will not get randomly deleted anymore either.

**Comment by kaisersly** [comment]
> Uh no idea if this is obviously the same issue or if this issue is generic, but I got this issue with the same error message OP mentioned probably because I uninstalled `xrdp` by accident. Probably via the ubuntu prompt to uninstall unneeded dependencies.
> 
> Fix was this, lol:
> 
> # inside the wsl2
> sudo apt install xrdp
> # make sure it's running
> sudo service xrdp status
> 
> Windows:
> 
> wsl --shutdown
> 
> That's it. Pretty sure that once you manually install `xrdp` like this it will not get randomly deleted anymore either.

I was in the same situation and your suggestion fixed the problem. It should be included in the docs IMO.

Thank you!
